src/client/Microsoft.Identity.Client/ManagedIdentity/ServiceFabricManagedIdentitySource.cs

```diff
@@ -42,21 +42,7 @@ public static AbstractManagedIdentity Create(RequestContext requestContext)
             return new ServiceFabricManagedIdentitySource(requestContext, endpointUri, EnvironmentVariables.IdentityHeader);
         }
 
-        internal override HttpClient GetHttpClientWithSslValidation(RequestContext requestContext)
-        {
-            if (_httpClientLazy == null)
-            {
-                _httpClientLazy = new Lazy<HttpClient>(() =>
-                {
-                    HttpClientHandler handler = CreateHandlerWithSslValidation(requestContext.Logger);
-                    return new HttpClient(handler);
-                });
-            }
-
-            return _httpClientLazy.Value;
-        }
-
-        internal HttpClientHandler CreateHandlerWithSslValidation(ILoggerAdapter logger)
+        internal override HttpClientHandler GetHttpClientHandlerWithSslValidation(ILoggerAdapter logger)
```

Comment:
Cool, so this fixes the other bug that we found, around SF changing the thumbprint.

Reply:
Yes this will also handle the case when a server cert is rotated. We no longer will cache the http client